hal/network/lwip/wiznet/wiznetif_config.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2023 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "wiznetif_config.h"
+
+#include "logging.h"
+LOG_SOURCE_CATEGORY("net.en.cfg")
+
+#include "system_error.h"
+
+#include "check.h"
+#include "delay_hal.h"
+
+#if HAL_PLATFORM_NCP
+#include "system_cache.h"
+#endif // HAL_PLATFORM_NCP
+
+namespace particle {
+
+namespace net {
+
+using namespace particle::services;
+
+WizNetifConfig::WizNetifConfig() :
+        initialized_(false) {
+}
+
+void WizNetifConfig::init() {
+    memset(&wizNetifConfigData_, 0, sizeof(wizNetifConfigData_));
+    recallWizNetifConfigData();
+    if (validateWizNetifConfigData() != SYSTEM_ERROR_NONE) {
+        recallWizNetifConfigData();
+    }
+    initialized_ = true;
+}
+
+// void WizNetifConfig::logWizNetifConfigData(WizNetifConfigData& configData) {
+//     LOG(INFO, "wizNetifConfigData size:%u cs_pin:%u reset_pin:%u int_pin:%u",
+//             configData.size,
+//             configData.cs_pin,
+//             configData.reset_pin,
+//             configData.int_pin);
+// }
+
+int WizNetifConfig::saveWizNetifConfigData() {
+    WizNetifConfigData tempData = {};
+    int result = SystemCache::instance().get(SystemCacheKey::WIZNET_CONFIG_DATA, (uint8_t*)&tempData, sizeof(tempData));
+    if (result != sizeof(tempData) ||
+            /* memcmp(tempData, wizNetifConfigData_, sizeof(tempData) != 0)) { // not reliable */
```

Comment:
Could you elaborate it?

Reply:
IIRC, for whatever reason it wasn't detecting changes in the structure.  This was on the FOTA PR though.  I left it here, but didn't try it again.  Maybe it would work on this simpler struct.

Reply:
There is a typo in there `, sizeof(tempData) != 0))` vs `, sizeof(tempData)) != 0)

Reply:
Hah, nice!  I'll convert to this method as it's simpler.